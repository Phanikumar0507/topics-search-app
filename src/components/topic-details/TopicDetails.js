import React, { useState } from "react";
import RelatedTopics from "./RelatedTopics";
import Topic from "./Topic";
import "./TopicDetails.scss";

const TopicDetails = ({ repoNode }) => {
  const [showRelated, setShowRelated] = useState(false);

  return (
    <>
      <Topic
        topic={repoNode.topic}
        setShowRelated={() => setShowRelated(true)}
      />
      {showRelated && (
        <RelatedTopics
          repoNode={repoNode}
          setShowRelated={() => setShowRelated(false)}
        />
      )}
    </>
  );
};

export default TopicDetails;
