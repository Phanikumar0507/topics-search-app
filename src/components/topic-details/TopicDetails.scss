.topic {
    display: flex;
    justify-content: space-between;
    padding: 16px;
    border: 1px solid #ddd;
    cursor: pointer;
  }
  
  .topic:hover {
    background-color: #f5f5f5;
  }
  
  .topic-name {
    font-weight: 600;
  }
  
  .topic-star-count {
    background-color: #607d8b;
    color: #fff;
    padding: 2px 4px;
    border-radius: 4px;
    border: 1px solid #455a64;
  }
  
  .modal-popup {
    /* Hidden by default */
    position: fixed; /* Stay in place */
    z-index: 1; /* Sit on top */
    left: 0;
    top: 0;
    width: 100%; /* Full width */
    height: 100%; /* Full height */
    overflow: auto; /* Enable scroll if needed */
    background-color: rgb(0, 0, 0); /* Fallback color */
    background-color: rgba(0, 0, 0, 0.4); /* Black w/ opacity */
  }
  
  /* Modal Content/Box */
  .modal-content {
    background-color: #fefefe;
    margin: 10% auto; /* 15% from the top and centered */
    border: 1px solid #888;
    width: 80%; /* Could be more or less, depending on screen size */
  }
  
  .modal-header {
    display: flex;
    justify-content: space-between;
    padding: 16px;
    background: #f5f5f5;
  }
  
  .close-btn {
    font-size: 24px;
    background-color: red;
    color: #fff;
    border: 1px solid red;
    border-radius: 50%;
    height: 30px;
    width: 30px;
    text-align: center;
    cursor: pointer;
  }
  
  .topics-list {
      padding: 16px;
  }
  