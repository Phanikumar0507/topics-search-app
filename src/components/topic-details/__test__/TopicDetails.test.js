import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import TopicDetails from "../topicDetails";

describe("Topic Details Component", () => {
  it("should render without crash", () => {
    const topic = { name: "vue", stargazerCount: 999 };
    const repoNode = {
      topic: topic,
    };
    const component = shallow(<TopicDetails repoNode={repoNode} />);
    expect(toJSON(component)).toMatchSnapshot();
  });
});
