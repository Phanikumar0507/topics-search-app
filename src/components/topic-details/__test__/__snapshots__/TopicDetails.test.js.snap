// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Details Component should render without crash 1`] = `
<Fragment>
  <Topic
    setShowRelated={[Function]}
    topic={
      Object {
        "name": "vue",
        "stargazerCount": 999,
      }
    }
  />
</Fragment>
`;
